op-batcher/batcher/driver.go

```diff
@@ -264,49 +262,54 @@ func (l *BatchSubmitter) calculateL2BlockRangeToStore(ctx context.Context) (eth.
 func (l *BatchSubmitter) loop() {
 	defer l.wg.Done()
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	ticker := time.NewTicker(l.PollInterval)
 	defer ticker.Stop()
 	for {
 		select {
 		case <-ticker.C:
 			l.loadBlocksIntoState(l.ctx)
+			l.publishStateToL1(ctx)
+		case <-l.done:
+			return
+		}
+	}
+}
+
+// publishStateToL1 loops through the block data loaded into `state` and
+// submits the associated data to the L1 in the form of channel frames.
+func (l *BatchSubmitter) publishStateToL1(ctx context.Context) {
+	for {
+		l1tip, err := l.l1Tip(ctx)
+		if err != nil {
+			l.log.Error("Failed to query L1 tip", "error", err)
+			return
+		}
```

Comment:
```suggestion
		l1ctx, cancel := context.WithTimeout(ctx, networkTimeout)
		l1tip, err := l.l1Tip(ctx)
		cancel()
		if err != nil {
			l.log.Error("Failed to query L1 tip", "error", err)
			return
		}
```
Or a broken l1 client connection might block very long here.

I think @trianglesphere is implementing more elaborate internal timeout handling in the transaction manager below, so should be fine below to just pass the `ctx` there.

Reply:
This now supports passing in a context to the `Stop()` method, which can be used to short-circuit any long blocking client calls, courtesy of @mdehoog